cpp/ql/src/semmle/code/cpp/PrintAST.qll

```diff
@@ -191,50 +181,37 @@ abstract class ASTNode extends PrintASTNode, TASTNode {
 class ExprNode extends ASTNode {
   Expr expr;
 
-  ExprNode() {
-    expr = ast
-  }
+  ExprNode() { expr = ast }
 
   override ASTNode getChild(int childIndex) {
     result.getAST() = expr.getChild(childIndex).getFullyConverted()
   }
 
   override string getProperty(string key) {
-    result = super.getProperty(key) or
-    (
-      key = "Value" and
-      result = getValue()
-    ) or
-    (
-      key = "Type" and
-      result = expr.getType().toString()
-    ) or
-    (
-      key = "ValueCategory" and
-      result = expr.getValueCategoryString()
-    )
-  }
-
-  string getValue() {
-    result = expr.getValue()
-  }
+    result = super.getProperty(key)
+    or
+    key = "Value" and
+    result = qlClass(expr) + getValue()
```

Comment:
```suggestion
    result = getValue()
```
Printing the class of the expression itself is redundant here. It's already printed by `toString()` on the expression object.

Reply:
The expression object will print the likes of `... + ...` (is this what you meant?), but again, my objective here is to print the name of the QL class that is being used.  There are a ton of `...Expr` nodes and it would be helpful to the novice (and not so novice) QL developer to see what they are.  So I'd vote to keep the `qlClass(expr)` in.